source/PyMaterialX/PyMaterialXGenShader/PyGenContext.cpp

```diff
@@ -0,0 +1,27 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <PyMaterialX/PyMaterialX.h>
+
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/HwShaderGenerator.h>
+
+namespace py = pybind11;
+namespace mx = MaterialX;
+
+void bindPyGenContext(py::module& mod)
+{
+    py::class_<mx::GenContext, mx::GenContextPtr>(mod, "GenContext")
+        .def(py::init<mx::ShaderGeneratorPtr>())
+        .def("getShaderGenerator", &mx::GenContext::getShaderGenerator)
+        .def("getOptions", static_cast<mx::GenOptions& (mx::GenContext::*)()>(&mx::GenContext::getOptions), py::return_value_policy::reference)
+        .def("registerSourceCodeSearchPath", static_cast<void (mx::GenContext::*)(const std::string&)>(&mx::GenContext::registerSourceCodeSearchPath))
+        .def("registerSourceCodeSearchPath", static_cast<void (mx::GenContext::*)(const mx::FilePath&)>(&mx::GenContext::registerSourceCodeSearchPath))
+        .def("registerSourceCodeSearchPath", static_cast<void (mx::GenContext::*)(const mx::FileSearchPath&)>(&mx::GenContext::registerSourceCodeSearchPath))
+        .def("findSourceCode", &mx::GenContext::findSourceCode)
+        .def("sourceCodeSearchPath", &mx::GenContext::sourceCodeSearchPath)
+        .def("addNodeImplementation", &mx::GenContext::addNodeImplementation)
```

Comment:
Do we need to expose add and find implementations?

Reply:
Not sure, @niklasharrysson?

Reply:
I agree those should be removed.